### PR TITLE
Add PKLot dataset

### PR DESCRIPTION
**Pull request by esnosy**
# Add Dataset

* Dataset Name: PKLot
* Issue Reference: https://github.com/tensorflow/datasets/issues/4085

## Description

Parking lots images database for empty/occupied parking spot classification

## Checklist

* [X] Address all TODO's
* [X] Add alphabetized import to subdirectory's `__init__.py`
* [X] Run `download_and_prepare` successfully
* [X] Add [checksums file](https://www.tensorflow.org/datasets/add_dataset#2_run_download_and_prepare_locally)
* [X] Properly cite in `BibTeX` format
* [ ] Add passing test(s)
* [ ] Add test data
* [ ] If using additional dependencies (e.g. `scipy`), use [lazy_imports](https://www.tensorflow.org/datasets/add_dataset#extra_dependencies) (if applicable)
* [ ] Add data generation script (if applicable)
* [X] [Lint](https://www.tensorflow.org/datasets/add_dataset#5_check_your_code_style) code

### CODE REVIEW

**Comment by iyadahmed** [comment]
any chance to get this in?

File: tensorflow_datasets/image_classification/pklot/pklot.py [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+      if not weather_dir_entry.is_dir():
+        continue
+
+      for date_dir_entry in os.scandir(weather_dir_entry):
+        if not date_dir_entry.is_dir():
+          continue
+
+        key_prefix = subdir + "_" + weather_dir_entry.name + "_" + date_dir_entry.name + "_"
+
+        for label_dir_entry in os.scandir(date_dir_entry):
```
Comment (fineguy) [inline tensorflow_datasets/image_classification/pklot/pklot.py:92]:
```suggestion
        for label in _LABELS:
          img_dir = date_dir / label

```

File: tensorflow_datasets/image_classification/pklot/pklot.py [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+            self._generate_examples(path, "PUC"),
+    }
+
+  def _generate_examples(self, path: Path, subdir: str):
+    """Yields examples."""
+    for weather_dir_entry in os.scandir(path / subdir):
+      if not weather_dir_entry.is_dir():
+        continue
+
+      for date_dir_entry in os.scandir(weather_dir_entry):
```
Comment (fineguy) [inline tensorflow_datasets/image_classification/pklot/pklot.py:86]:
```suggestion
      for date_dir in weather_dir.iterdir():
```

File: tensorflow_datasets/image_classification/pklot/pklot.py [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+_EMPTY_LABEL_DIR_NAME = _EMPTY_LABEL
+_OCCUPIED_LABEL_DIR_NAME = _OCCUPIED_LABEL
+_LABEL_DIR_NAME_TO_LABEL_MAP = {
+    _EMPTY_LABEL_DIR_NAME: _EMPTY_LABEL,
+    _OCCUPIED_LABEL_DIR_NAME: _OCCUPIED_LABEL
+}
+
+
+class Pklot(tfds.core.GeneratorBasedBuilder):
+  """DatasetBuilder for pklot dataset."""
```
Comment (fineguy) [inline tensorflow_datasets/image_classification/pklot/pklot.py:34]: The dataset provides original and segmented images that have a slightly different directory structure:
https://www.tensorflow.org/datasets/add_dataset?hl=en#dataset_configurationvariants_tfdscorebuilderconfig

**Review by fineguy — CHANGES_REQUESTED** [review]
Thanks for your contribution! 

I'll be glad to help you get this merged, I left a few comments with some ideas for improvement.

File: tensorflow_datasets/image_classification/pklot/pklot.py (hunk 3 above)
Comment (iyadahmed) [reply tensorflow_datasets/image_classification/pklot/pklot.py:34]: I just saw GitHub thanks a ton for reviewing this!, Will get to it asap 🙏

**Pull request closed without merging**